cloud/pkg/dynamiccontroller/application/application.go

```diff
@@ -253,7 +265,9 @@ type Agent struct {
 
 // edged config.Config.HostnameOverride
 func NewApplicationAgent(nodeName string) *Agent {
-	return &Agent{nodeName: nodeName}
+	agent := Agent{nodeName: nodeName}
+	go agent.GC()
+	return &agent
```

Comment:
Since there is no limit to the number of `NewApplicationAgent` calls, this may lead to `goroutine leak`.
But currently it will only be used once(https://github.com/kubeedge/kubeedge/search?q=NewApplicationAgent)，so  a TODO or comment can be added to remind other users.

If you want to fix this, you can use the global gc to clean up all the `Applications` of the created agent，
or add a shutdown signal to the agent

Reply:
fixed  PTAL

Reply:
@Rachel-Shao seems not fix the problem of `goroutine leak` if the number of `NewApplicationAgent` calls increase. I prefer to that agent is shared, so how about setting a global agent?
```go
var defaultAgent *Agent
var once sync.Once
func NewApplicationAgent() *Agent{
  once.Do(){
     ...
  }
}

Reply:
Modified:)